Save product details to database

kamrulbari14 opened the pull request:
-- ApiController
-- BeanConfig
-- ProductController
-- BaseDto
-- ProductDto
-- Make created at not updatable in BaseEntity
-- Rename sellablePrice to saleablePrice in Product
-- ProductRepository
-- ProductService and ProductServiceImpl

kamrulbari14 committed:
Save product details to database
-- ApiController
-- BeanConfig
-- ProductController
-- BaseDto
-- ProductDto
-- Make created at not updatable in BaseEntity
-- Rename sellablePrice to saleablePrice in Product
-- ProductRepository
-- ProductService and ProductServiceImpl

akhter-rakib commented on src/main/java/com/inventory/demo/dto/ProductDto.java:
```diff
@@ -0,0 +1,15 @@
+    private String batchId;
+    private Double originalPrice;
+    private Double salablePrice;
+    private Double totalQuantity;
```
Need to add the Bean validation for user Input

kamrulbari14 replied:
Required Validation added

akhter-rakib commented on src/main/java/com/inventory/demo/entity/Product.java:
```diff
@@ -10,8 +10,8 @@
     private Double originalPrice;
-    private Double sellablePrice;
+    private Double salablePrice;
     private Double totalQuantity;
```
Need to add the JPA Bean validation for user Input

kamrulbari14 replied:
Required Validation added

akhter-rakib commented on src/main/java/com/inventory/demo/service/impl/ProductServiceImpl.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Override
+    public Response saveProduct(ProductDto productDto) {
+        Product product = productRepository.findByBatchIdAndIsActive(productDto.getBatchId(), Boolean.TRUE);
```
Can ignore the validation with batch. Apply product name validation only

kamrulbari14 replied:
Removed the batch Id property and the validation

akhter-rakib reviewed (changes requested):
Please have a look at the comment and change accordingly.

kamrulbari14 committed:
Validation logic added for saving data
-- ProductDto
-- Product
-- isActive datatype changed in BaseEntity and BaseDto
-- ProductController
-- BatchId validation removed before saving data in ProductServiceImpl and ProductRepository

akhter-rakib committed:
Product save feature

-- added custom validation
-- use condition to check salable policy
